### PR TITLE
(FM-7681) puppetserver_gem - match gem documentation params 

### PR DESCRIPTION
**Pull request by tkishel**
With this commit ...

The `--no-rdoc` and `--no-ri` options are replaced with `--no-document`.

Minor changes related to the above are synced from the gem provider.

The license and metadata are updated to correct Forge score issues.

### CODE REVIEW

**Comment by tkishel** [comment]
I can split the license and metadata changes out of the single commit into a maint commit, if necessary.

**Comment by tkishel** [comment]
This PR includes the same changes as in the recently merged PR https://github.com/puppetlabs/puppet/pull/7320 triggered by https://github.com/bundler/bundler/releases/tag/v2.0.0

(The --no-rdoc and --no-ri options were deprecated and are replaced with --no-document in version 3.x of the gem command.)

**Comment by tkishel** [comment]
@sebastian-miclea This is related to PUP-9395, could you review?

**Comment by justinstoller** [comment]
This requires the rubygems version to be greater than 2.0, right?

**Comment by tkishel** [comment]
Yes `--document` introduced here:

https://github.com/rubygems/rubygems/blob/master/History.txt#L2068

and `--rdoc` and `ri` removed here:

https://github.com/rubygems/rubygems/blob/master/History.txt#L43

**Comment by tkishel** [comment]
We document compatibility with:

```
Puppet Enterprise 2019.1.x, 2019.0.x, 2018.1.x, 2017.3.x, 2017.2.x, 2017.1.x, 2016.5.x, 2016.4.x
Puppet >=4.0.0 < 7.0.0
```
```
[root@pe-2016415-master ~]# /opt/puppetlabs/puppet/bin/gem --version
2.2.5
```

So we are okay, yes?

**Comment by justinstoller** [comment]
I think we're using (sometimes) `/opt/puppetlabs/bin/puppetserver gem` vs `/opt/puppetlabs/puppet/bin/gem` which might be a different gem version. I'm surprised to find that even though we're shipping a version of JRuby that uses Ruby 1.9 we use Rubygems within it that have 2.x versions, ie:
```
[root@erxvbqh08pfgqls puppet-enterprise-2016.4.0-el-7-x86_64]# /opt/puppetlabs/bin/puppetserver gem --version
2.4.8
```

So, yeah, this is fine, :disappear:

**Comment by justinstoller** [comment]
I think @Sharpie @tphoney and @pmcmaw where the stakeholders in puppetserver_gem improvements if they want to review/merge.

**Comment by tkishel** [comment]
Since we implement a shared gempath (SERVER-2133) between puppet and puppetserver, I expect both (versions of) the gem command are close and compatible.

**Comment by justinstoller** [comment]
That was implemented in upstream FOSS with PE 2018.1 in mind (which also moved customers to JRuby 9k). My point was I wouldn't make that assumption for people running JRuby 1.7 or FOSS versions prior to 5.3, though I'm happily surprised that it works.

**Comment by tkishel** [comment]
Thanks for the clarification!

**Comment by tphoney** [comment]
👍

**Comment by er0ck** [comment]
@tphoney i think server is cool with this.  do you want us to merge it, or are you all going to do that?

File: metadata.json [hunk 1]
```diff
@@ -3,10 +3,60 @@
...
+        "6",
+        "7"
+      ]
+    },
+    {
+      "operatingsystem": "RedHat",
+      "operatingsystemrelease": [
+        "5",
+        "6",
+        "7"
```
Comment (Sharpie) [inline metadata.json]: While we're here, should we proactively add `"8"` to the RedHat, CentOS, OracleLinux lists?

Comment (tkishel) [reply metadata.json]: Word. Added a commit!

**Comment by vchepkov** [comment]
Works on Puppet 6.5.0 for me, thank you, @tkishel !

**Comment by tkishel** [comment]
Thanks for bringing it to our attention! I overlooked this since the inheritance spans repositories.

**Comment by er0ck** [comment]
@justinstoller is it a good idea for us to merge this?  i didn't hear back from @tphoney

**Comment by justinstoller** [comment]
Happy to merge, though I'm not aware of how to release these bits. I'll leave that to modules/ecosystem team.

**Comment by tkishel** [comment]
Thank you!

**Pull request merged**